extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -36,57 +35,100 @@ public DevConsoleTemplateInfoBuildItem collectAdditionalEndpoints(
         });
 
-        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", knownPaths);
+        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", staticResourceInfo);
     }
 
-    private void collectKnownPaths(Path resource, Set<String> staticResources) {
+    //change collectKnownPaths to build those(Resource Paths) items and fill the map
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
 
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+                        throws IOException {
+                    Path relativePathOfDir = resource.relativize(dir);
+                    Path dirName = dir.getFileName();
+                    Path parentOfDir = dir.getParent().getFileName();
+
+                    if (!staticResourceInfo.resourceMap.containsKey(relativePathOfDir)) {
+                        StaticResourceInfo.StaticFile newStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(),
+                                attrs.isDirectory(), null);
+                        staticResourceInfo.resourceMap.put(relativePathOfDir, newStaticFile);
+
+                        // add newStaticFile in list of child of parentOfDir
+                        if (staticResourceInfo.resourceMap.containsKey(parentOfDir)) {
+                            StaticResourceInfo.StaticFile valueOfParentKey = staticResourceInfo.resourceMap.get(parentOfDir);
+                            List<StaticResourceInfo.StaticFile> retrieveChildListFromParent = valueOfParentKey.children;
+                            retrieveChildListFromParent.add(newStaticFile);
+                            staticResourceInfo.resourceMap.put(parentOfDir, new StaticResourceInfo.StaticFile(
+                                    parentOfDir.toString(), true, retrieveChildListFromParent));
                         } else {
-                            String parentString = parent.toString();
-                            if (!parentString.startsWith("/")) {
-                                parentString = "/" + parentString;
-                            }
-                            staticResources.add(parentString + "/");
+                            List<StaticResourceInfo.StaticFile> childList = new ArrayList<>();
```

Comment:
Do not consider the case where the parent is missing: you're adding those parents yourself before you visit the children, so they will exist. Just throw an exception if they don't because it's a bug.

Reply:
ok, adding `throw new NotFoundException("Parent folder not found!");`

Reply:
Oh @FroMage, I just checked on removing the else part, it is throwing exception "Parent Folder Not Found"! That means it's not getting parent folder 🤔 but it shouldn't happen.

![image](https://user-images.githubusercontent.com/46818757/106734337-eae4d880-6638-11eb-9401-895760943df0.png)

Reply:
It would help if you showed which parent folder is missing.

Reply:
I tried fixing it. 
Right now,  I am getting NullPointer Exception in `preVisitDirectory` on finding the relative parent path `resource.relativize(dir).getParent()`.

I think this is expected.

Let's say I start walking over files and folders from resources folder  i.e `deployment/src/main/resources`.

So for the 1st time inside `previsit`, 

Path resource = `deployment/src/main/resources`
Path dir = `deployment/src/main/resources`

since both will be same so `resource.relativize(dir)` is expected to be NULL and that's why `resource.relativize(dir).getParent()` is also NULL.

WDYT

Reply:
Yes you need to seed the stack with the original root folder `/`.